전남대 BE_서영우 3주차 과제 (2단계)

westzeroright opened the pull request:
객체 참조를 하고 싶었는데 잘 한 건지는 잘 모르겠습니다..

westzeroright committed:
chore: spring-gift-wishlist 레포지토리 소스코드 이전

westzeroright committed:
feat: 로그아웃한 계정이 기능 사용시 예외 처리

westzeroright committed:
chore: 패키지 이동

westzeroright committed:
refacotr: 별도의 예외 클래스 관리

westzeroright committed:
refacotr: 커스텀 예외 클래스 사용

westzeroright committed:
refacotr: InvalidProduct 커스텀 예외 클래스 적용

westzeroright committed:
refacotr: 시간과 날짜 표현 방식 변경

westzeroright committed:
refacotr: Prodcut 도메인에 관해 서비스 계층을 둠

westzeroright committed:
refacotr: Member 도메인에 관해 서비스 계층을 둠

westzeroright committed:
refacotr: Wishlist 도메인에 관해 서비스 계층을 둠

westzeroright committed:
refacotr: LogoutToken 도메인에 관해 서비스 계층을 둠

westzeroright committed:
refacotr: JdbcClient 대신 JdbcTemplate 사용

westzeroright committed:
style: 불필요한 주석 및 임포트문 제거

westzeroright committed:
test: ProductControllerTest 테스트 코드 작성

westzeroright committed:
refactor: setter 제거

westzeroright committed:
refactor: 이미 존재하는 회원정보에 대한 예외처리를 GlobalExceptionHandler에서 처리

westzeroright committed:
refactor: 회원 조회 실패에 대한 예외처리를 GlobalExceptionHandler에서 처리

westzeroright committed:
chore: jpa 의존성 추가

westzeroright committed:
feat: 엔티티 클래스 정의

westzeroright committed:
feat: JPA 사용 시 동작 쿼리 로그로 확인

westzeroright committed:
feat: MYSQL Dialect 사용

westzeroright committed:
feat: 엔티티 클래스 수정

많이 졸린가봅니다..

westzeroright committed:
refactor: product 코드 jpa로 리팩토링

westzeroright committed:
refactor: Delete ProductDao

jpa로 리팩토링

westzeroright committed:
refactor: product 예외 처리 로직

westzeroright committed:
style: 불필요한 주석 제거

westzeroright committed:
refactor: member(회원가입,로그인) JPA로 리팩토링

westzeroright committed:
refactor: Delete MemberDao

westzeroright committed:
refactor: wishlist JPA 리팩토링

westzeroright committed:
style: 공백 제거

westzeroright committed:
refactor: 필요한 메서드 복구

westzeroright committed:
refactor: 필요한 메서드 추가

westzeroright committed:
refactor: JwtTokenUtil 빈으로 등록하지 않기

westzeroright committed:
refactor: decodeJwt 메서드 static으로 설정

westzeroright committed:
refactor: Delete WishlistDao

westzeroright committed:
style: 불필요한 주석 제거

westzeroright committed:
style: 불필요한 공백줄 제거

westzeroright committed:
refactor: Delete schema.sql

JPA로 리팩토링 했기 때문에 불필요하므로 제거

westzeroright committed:
test: @DataJpaTest 실행

westzeroright committed:
fix: application.properties 수정

westzeroright committed:
refactor: wishlist id getter 필요해서 추가

westzeroright committed:
docs: 과제 수행 목록 작성

westzeroright committed:
style: 불필요한 주석 제거

westzeroright committed:
refacotr: 삭제했던 blacklist 테이블 복구

westzeroright committed:
refacotr: 참조관계 wishlist 추가

westzeroright committed:
refacotr: 참조관계 wishlist 추가

westzeroright committed:
refacotr: 장바구니에서 삭제

memberId와 productId가 모두 일치한 행을 삭제합니다

westzeroright committed:
style: 불필요한 주석 제거

westzeroright committed:
refactor: WishlistRepositoryTest 수정

westzeroright committed:
style: 불필요한 주석 제거

westzeroright committed:
refactor: 디버깅

westzeroright committed:
refactor: 디버깅

westzeroright committed:
refactor: ?를 컨트롤러 전체적으로 제거하기

westzeroright committed:
refactor: findByEmail도 Optional로 감싸기

westzeroright committed:
refactor: 사용하지 않는 메서드 제거

MemberService.getMemberByEmail

westzeroright committed:
refactor: 사용하는 메서드 다시 추가..

MemberService.getMemberByEmail

westzeroright committed:
refactor: RuntimeException을 상속받도록 수정

westzeroright committed:
refactor: Optional을 사용하므로 if문 없앰

westzeroright committed:
refactor: 데이터를 주고받음에 있어서 DTO를 사용하도록 함

westzeroright committed:
refactor: Optional 사용

pkeugine commented on README.md:
```diff
@@ -1 +1,11 @@
-# spring-gift-jpa
\ No newline at end of file
+# spring-gift-jpa
```
conflict 함 봐주세요!

pkeugine commented on src/main/java/gift/member/Member.java:
```diff
@@ -1,28 +1,37 @@
-    public Member(String id, String name, String email, String password, boolean role) {
-        this.id = id;
-        this.name = name;
+    protected Member() {}
```
외부에서 사용하지 않을 생성자 `protected` 처리 👍

pkeugine commented on src/main/java/gift/product/Product.java:
```diff
@@ -1,21 +1,42 @@
+   @Column(name = "image_url", nullable = false)
+   private String imageUrl;
+
+   protected Product() {}
+
+    public Product(String name, int price, String imageUrl) {
```
코드 컨벤션 한 번만 확인해주세요!
어쩌다가 이렇게 인덴트가 만들어진...거죠??

pkeugine commented on src/main/java/gift/product/ProductController.java:
```diff
@@ -27,64 +27,39 @@ public ProductController(ProductService productService) {
-            productRequestDto.url()
-        );
-
-        productService.postProduct(product);
 
         return new ProductResponseDto(
-            product.getId(),
-            product.getName(),
-            product.getPrice(),
-            product.getUrl()
+            product.get().getId(),
+            product.get().getName(),
+            product.get().getPrice(),
+            product.get().getImageUrl()
         );
     }
```
Optional 에 어떤 메서드가 있는지 한 번 알아보고,
if 문과 `get()` 을 없앤 로직을 한 번 작성해보아요

pkeugine commented on src/main/java/gift/wishlist/Wishlist.java:
```diff
@@ -0,0 +1,74 @@
+    @ManyToOne
+    private Product product;
+
+    public Wishlist() {
+    }
```
위에 칭찬한게 여긴 안되어있네요 ㅠ

pkeugine commented on src/main/java/gift/wishlist/Wishlist.java:
```diff
@@ -0,0 +1,74 @@
+        return product;
+    }
+
+    //    @Column(name = "member_id", nullable = false)
+//    private Long memberId;
+//
+//    @Column(name = "product_id", nullable = false)
+//    private Long productId;
+//
+//    protected Wishlist() {}
+//
+//    public Wishlist(Long memberId, Long productId) {
+//        this.memberId = memberId;
+//        this.productId = productId;
+//    }
+//
+//    public Long getId() {
+//        return id;
+//    }
+//
+//    public Long getMemberId() {
+//        return memberId;
+//    }
+//
+//    public Long getProductId() {
+//        return productId;
+//    }
+}
```
할많하않

westzeroright replied:
🥲🌧️

pkeugine commented on src/main/java/gift/product/ProductController.java:
```diff
@@ -27,64 +27,39 @@ public ProductController(ProductService productService) {
-    public Optional<ProductResponseDto> getProduct(@PathVariable Long id) {
-        Optional<ProductResponseDto> product = productService.getProductById(id);
+    public ProductResponseDto getProduct(@PathVariable Long id) {
+        Optional<Product> product = productService.getProductById(id);
```
아 그리고 아예 controller 가 domain 에 의존하지 않도록 해볼까요?
이 코멘트는 왜 남겼는지 한 번 고민해보고 이유를 알아보도록 해요!
(코멘트 남겨주세요)

westzeroright replied:
객체지향의 캡슐화에 따라 도메인 내부 구현이 외부에 노출되지 않도록 하기 위해서 제안해주신 바를 따라야할 것 같습니다..!

pkeugine commented on src/test/java/gift/wishlist/WishlistRepositoryTest.java:
```diff
@@ -0,0 +1,112 @@
+
+        assertAll(
+            () -> assertThat(wishlist.size()).isEqualTo(2)
+        );
+
+    }
+
+}
```
딱히 모든 메서드 끝부분에 공백을 추가하지는 않으시는거 같으니
이런건 싹 다 제거해보도록 하죠.
이와 관련된 코멘트는 이제 안 받는걸 목표로 해요!

westzeroright replied:
노력해보겠습니다🥹

pkeugine commented on src/test/java/gift/product/ProductRepositoryTest.java:
```diff
@@ -0,0 +1,87 @@
+
+        List<Product> productList = productRepository.findAll();
+
+        assertAll(
+            () -> assertThat(productList.size()).isEqualTo(3)
+        );
```
`assertAll` 이 필요없는 곳임에도 사용한건

1. 통일성을 위한건가요
2. 아니면 복붙의 영향인가요?

westzeroright replied:
통일성을 위해 사용하였습니다

westzeroright replied:
혹시 불필요한 코드로 간주되어 지양될까요?

pkeugine replied:
음 아뇨 그냥 궁금했어요 ㅋㅋㅋ
통일성 때문이었다면 저도 존중합니다!!!

pkeugine commented on src/test/java/gift/product/ProductRepositoryTest.java:
```diff
@@ -0,0 +1,87 @@
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Test
```
`@DisplayName` 을 활용해서 테스트에 이름을 붙여보죠.
테스트 클래스에도 붙일 수 있어요!

pkeugine commented on src/main/java/gift/wishlist/WishlistService.java:
```diff
@@ -1,32 +1,71 @@
-        wishlistDao.deleteWish(productId);
+    @Transactional
+    public HttpEntity<String> deleteWishlist(Long productId, Long memberId) {
+        List<Wishlist> w1 = wishlistRepository.findByProductId(productId);
+        List<Wishlist> w2 = wishlistRepository.findByMemberId(memberId);
```
변수에 의미있는 이름을 지어볼까요?

pkeugine reviewed (changes requested):
안녕하세요 영우님!
바로 step2 까지 빠르게 봤어요.
한 번 보시고, 반영 및 댓글 남겨주세요!
주말에도 고생 많으십니다 💪 💪 💪

westzeroright committed:
merge step1

westzeroright committed:
feat: @DisplayName 적용

westzeroright committed:
refactor: step2 코드 리뷰 피드백 반영

- conflict 해결 (step1 merge)
- 인덴트 수정 (인텔리제이와 친해지기..)
- Optional 메서드 활용 & if문과 get() 없애기
- 생성자 protected 처리
- controller가 도메인에 의존하지 않도록

westzeroright committed:
Merge branch 'westzeroright' into step2

westzeroright committed:
refactor: conflict 해결

westzeroright committed:
style: 메서드 끝부분 공백 제거 방식으로 통일

pkeugine reviewed (approved):
바로 다음 단계로 넘어가보아요!!! 👍👍
